core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -84,8 +84,15 @@ private[spark] class CoarseGrainedExecutorBackend(
       }
 
     case x: DisassociatedEvent =>
-      logError(s"Driver $x disassociated! Shutting down.")
-      System.exit(1)
+      if (x.remoteAddress == driver.anchorPath.address) {
+        logError(s"Driver $x disassociated! Shutting down.")
+        System.exit(1)
+      } else {
+        val localAddress = x.localAddress
+        val remoteAddress = x.remoteAddress
+        logWarning(s"Received irrelevant DisassociatedEvent [$localAddress]" +
+          s"${if (x.inbound) " <- " else " -> "}[$remoteAddress]")
```

Comment:
can you rewrite this line? take the if outside of string interpolation. it is too hard to read.